### ISSUE TITLE
FileLoggingHandler fails the first time it creates a log file

### ISSUE THREAD

**Issue by Plastonick** [simplesamlphp/simplesamlphp#1877]
**Describe the bug**

When `FileLoggingHandler` first tries to create the log file, it necessarily fails since symfony/filesystem's `touch` method returns void, whereas `FileLoggingHandler` uses it as though it returns boolean.

See:

```php
$file = new File($this->logFile, false);
// Suppress E_WARNING if not exists
if (@$this->fileSystem->exists($this->logFile)) {
    if (!$file->isWritable()) {
        throw new CannotWriteFileException(
            sprintf("Could not write to logfile: %s", $this->logFile),
        );
    }

 // this is necessarily true, since `touch` returns void. `touch` throws an exception on failure.
} elseif (!$this->fileSystem->touch($this->logFile)) {
    throw new CannotWriteFileException(sprintf(
        "The logging directory is not writable for the web server user. Could not create logfile: %s",
        $this->logFile,
    ));
}
```

**To Reproduce**

1. Turn on filesystem logging in the configuration
2. pointing the logger at a file that doesn't yet exist
3. prompt it to log something, note it creates the log file, but then immediately throws an exception
4. subsequent attempts will succeed, since the file will exist

**Expected behavior**

it should not fail by default if the log file doesn't yet exist.

**Screenshots or logs**

n/a

**Additional context**

I believe simplesamlphp used to use PHP's `touch` method previously, which _does_ have a boolean return indicating success.

**Comment by tvdijen** [comment]
Closed in 49cef15 - will be part of the next bugfix-release.

**Comment by tvdijen** [comment]
Ehh, no.. Maybe I responded a bit too fast ;)  Fixed it in fe26133d2d341ada9e54dd384e89d50c9d181b7d
